### ISSUE TITLE
Chapter 4 - Implement algorithms  

### ISSUE THREAD

**Issue by sambitdash** [sambitdash/AIMACore.jl#4]
The following algorithms are not implemented yet. 

1. AND-OR tree
2. Online DFS
3. LRTA

**Comment by chrishna1** [comment]
Alright. I'll start implementing these algorithms. Can I ask questions if I stuck somewhere ?

**Comment by sambitdash** [comment]
Of course, this is the best place for getting the clarifications and trail can be reviewed later as well.

**Comment by chrishna1** [comment]
Okay thanks.

**Comment by chrishna1** [comment]
In AND_OR_Graph_Search we have nested function call ,but since we pass problems in search followed by execute, how can I call AND_Search by passing the current state and action?

**Comment by chrishna1** [comment]
Or it can be solved if for this search algorithm I create a seperate helper AND_Search and OR_Search function which helps in the particular algorithm. What I mean is we can't use AND_Search or OR_Search as a complete algorithm as it is just a part of AND_OR_Search so I don't have to create separate composite type for the two and they won't be exported also.

**Comment by chrishna1** [comment]
Does the preceding comment makes sense?

**Comment by sambitdash** [comment]
Search strategies are struct and then the methods act on the objects. AND-OR-SEARCH will be the struct and then the methods will work on it as well as the Problems.

**Comment by Parth-DH** [comment]
I would like to work on this issue, if that's alright?

**Comment by sambitdash** [comment]
@Parth-DH You are most welcome.
